### PR TITLE
Exclude pkg and extern from codecov coverage gathering script

### PR DESCRIPTION
**Pull request by markuspf**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -135,7 +135,7 @@ script:
 after_script:
   - set -e
   - bash etc/ci-gather-coverage.sh
-  - bash <(curl -s https://codecov.io/bash)
+  - bash <(curl -s https://codecov.io/bash) -f "!./pkg/*" -f "!./extern/*"
```
Comment (fingolfin) [inline .travis.yml]: Ah, I did not try that. If this doesn't work, I wonder if the `-g GLOB` option might also be useful?

**Comment by fingolfin** [comment]
Unfortunately there is still garbage in the coverage for `io.c`, see https://codecov.io/gh/gap-system/gap/src/59778277d4b8ece2aee79af6741e25769aa39b3d/src/io.c

**Comment by markuspf** [comment]
yes. it comes from the gmp build...

**Comment by markuspf** [comment]
also the exclusion doesn't work on appveyor yet of course

File: .travis.yml [hunk 2]
```diff
@@ -135,7 +135,7 @@ script:
 after_script:
   - set -e
   - bash etc/ci-gather-coverage.sh
-  - bash <(curl -s https://codecov.io/bash)
+  - bash <(curl -s https://codecov.io/bash) -f "!./pkg/*" -f "!./extern/*" -f "!./build/*"
```
Comment (fingolfin) [inline .travis.yml]: Ahh, I bet I forgot to exclude `build` in past attempts, too, which might explain some things.

**Comment by markuspf** [comment]
aaand we got liftoff. The coverage of `io.c` still doesn't seem quite right, but it is now more sensible.

**Comment by markuspf** [comment]
now for some wrestling with bash... because why not.

**Comment by fingolfin** [comment]
Things are indeed improving. `io.c` coverage seems mostly OK, in that only lines that really are code seem to be marked as such (and not anymore stuff in the middle of comments etc.). But something is still off -- lots of code is not marked at all, even code that is preceded and followed by code marked as executable.

**Pull request merged**